polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -733,22 +810,73 @@ impl State {
 		);
 
 		if let Some(approval_entry) = candidate_entry.approval_entry(&block_hash) {
-			let required_tranches = approval_checking::tranches_to_approve(
-				approval_entry,
-				candidate_entry.approvals(),
-				tranche_now,
-				block_tick,
-				no_show_duration,
-				session_info.needed_approvals as _,
-			);
+			let TranchesToApproveResult { required_tranches, total_observed_no_shows } =
+				approval_checking::tranches_to_approve(
+					approval_entry,
+					candidate_entry.approvals(),
+					tranche_now,
+					block_tick,
+					no_show_duration,
+					session_info.needed_approvals as _,
+				);
 
-			let status = ApprovalStatus { required_tranches, block_tick, tranche_now };
+			let status = ApprovalStatus {
+				required_tranches,
+				block_tick,
+				tranche_now,
+				last_no_shows: total_observed_no_shows,
+			};
 
 			Some((approval_entry, status))
 		} else {
 			None
 		}
 	}
+
+	// Returns the approval voting params from the RuntimeApi.
+	#[overseer::contextbounds(ApprovalVoting, prefix = self::overseer)]
+	async fn get_approval_voting_params_or_default<Context>(
```

Comment:
This would be more idiomatic to return `Option<ApprovalVotingParams>`  and have `ApprovalVotingParams` implement `Default` rather than hardcoding the default impl inside this fn

Reply:
Easy to fix will do.